File: hw4/Calculator.fs
module hw4.Calculator

open hw4.Operation

let Calculate (operation:Operation) (val1: int) (val2: int) =
    match operation with
    | Operation.Add -> val1 + val2
    | Operation.Subtract -> val1 - val2
    | Operation.Divide -> val1/val2
    | _ -> val1*val2

File: hw8/hw8/Calculator.cs
namespace hw8
{
    public class Calculator : ICalculator
    {
        public double Calculate(double val1, Operation operation, double val2)
        {
            double result = 0;
            switch (operation)
            {
                case Operation.Add:
                    result = val1 + val2;
                    break;
                case Operation.Subtract:
                    result = val1 - val2;
                    break;
                case Operation.Multiply:
                    result = val1 * val2;
                    break;
                case Operation.Divide:
                    result = val1 / val2;
                    break;
            }
            return result;
        }
    }
}